src/colvarcomp_apath.cpp

```diff
@@ -0,0 +1,193 @@
+#if (__cplusplus >= 201103L)
+
+#include <numeric>
+#include <algorithm>
+#include <cmath>
+#include <cstdlib>
+#include <limits>
+
+#include "colvarmodule.h"
+#include "colvarvalue.h"
+#include "colvarparse.h"
+#include "colvar.h"
+#include "colvarcomp.h"
+
+colvar::aspathCV::aspathCV(std::string const &conf): CVBasedPath(conf) {
+    function_type = "aspathCV";
+    cvm::log(std::string("Total number of frames: ") + cvm::to_str(total_reference_frames) + std::string("\n"));
+    std::vector<cvm::real> p_weights(cv.size(), 1.0);
+    get_keyval(conf, "weights", p_weights, std::vector<cvm::real>(cv.size(), 1.0));
+    x.type(colvarvalue::type_scalar);
+    use_explicit_gradients = true;
+    std::vector<cvm::real> rmsd_between_refs(total_reference_frames - 1, 0.0);
+    distanceBetweenReferenceFrames(rmsd_between_refs);
+    cvm::real mean_square_displacements = 0.0;
+    for (size_t i_frame = 1; i_frame < total_reference_frames; ++i_frame) {
+        cvm::log(std::string("Distance between frame ") + cvm::to_str(i_frame) + " and " + cvm::to_str(i_frame + 1) + " is " + cvm::to_str(rmsd_between_refs[i_frame - 1]) + std::string("\n"));
+        mean_square_displacements += rmsd_between_refs[i_frame - 1] * rmsd_between_refs[i_frame - 1];
+    }
+    mean_square_displacements /= cvm::real(total_reference_frames - 1);
```

Comment:
At some point before this, an error should be raised if `total_reference_frames` is 1. I couldn't find it easily in the base class.